libs/libarchfpga/src/physical_types.h

```diff
@@ -1225,6 +1225,7 @@ struct t_pin_to_pin_annotation {
  *      parent_pb_graph_node  : parent pb graph node
  *      total_primitive_count : Total number of this primitive type in the cluster. If there are 10 ALMs per cluster
  *                              and 2 FFs per ALM (given the mode of the parent of this primitive) then the total is 20.
+ *      flat_site_index       : index of this primitive within its primitive type in this cluster; in [0,...,total_primitive_count]
```

Comment:
An example in the comment would be good that makes clear how this is used and how it differs from placement_index. (e.g. if you have 10 LUTs and 10 FFs in a hierarchy level would flat_site_index goes from 0 to 19?).  Explain what this is used for -- I believe it is only for reading in external flat placments so explain that and what it is useful for in that case.